tests/unit/Map.ts

```diff
@@ -0,0 +1,239 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import Map from 'src/Map';
+
+var map: Map<any, any>;
+
+registerSuite({
+	name: 'Map',
+
+	instantiation: {
+		'null data'() {
+			assert.doesNotThrow(function () {
+				map = new Map<number, string>(null);
+			});
+		},
+
+		'undefined data'() {
+			assert.doesNotThrow(function () {
+				map = new Map<number, string>(undefined);
+			});
+		},
+
+		'empty data'() {
+			assert.doesNotThrow(function () {
+				map = new Map<number, string>([]);
+			});
+		},
+
+		'array like data'() {
+			assert.doesNotThrow(function () {
+				map = new Map<number, string>({
+					length: 1,
+					0: [1337, 'bar']
+				});
+			});
+		}
+	},
+
+	set: {
```

Comment:
We should probably test that `size` updates, either in one of these tests, or in a separate test after the `set` tests.